lib/hook.js

```diff
@@ -57,40 +73,37 @@ Hook.HOOK_PRIORITY_DEFAULT = 1000;
  * Scans the specified path for hooks. Each hook is identified by a JavaScript
  * file.
  * @param {String} dir - A directory or JavaScript file to be scanned.
+ * @param {RegExp} include - Regular expression to match for included filenames.
+ * @param {RegExp} exclude - Regular expression to match for excluded filenames.
  */
-Hook.prototype.scanHooks = function scanHooks(dir) {
+Hook.prototype.scanHooks = function scanHooks(dir, include, exclude) {
 	if (!this.hooks.scannedPaths[dir] && fs.existsSync(dir)) {
-		var jsfile = /\.js$/,
-			ignore = /^[._]/,
-			isDir = fs.statSync(dir).isDirectory();
+		const fileList = findInDir(dir, include, exclude);
```

Comment:
This is probably a bad idea. You are recursively walking all subdirectories running a regex against each filename. What if the plugin directory has a `node_modules` directory? What if someone sets their `paths.hooks` to `/Users`? This code is going to scan this directory every time you run the CLI and it's going to slow things down and nobody wants that. Adding a `depth` would improve the situation.

Reply:
I could see someone wanting to install a plugin via npm which would make it exist in the `node_modules` directory